src/datarobot_asgi_middleware/__init__.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2025 DataRobot, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+from starlette.types import ASGIApp, Receive, Scope, Send
+
+
+class DataRobotASGIMiddleware:
+    """
+    Middleware to augment ASGI applications run by DataRobot Custom Applications.
+
+    It routes root URL requests from Kubernetes to the base /health URL to support
+    more robust health checks than just loading the root URL.
+    """
+
+    def __init__(self, app: ASGIApp, use_health: bool = False):
+        self.app = app
+        self.use_health = use_health
+        # Get the script name from the environment variable to know what the internal
+        # load balancer is using as the prefix for the request.
+        self.internal_prefix = os.getenv("SCRIPT_NAME", None)
```

Comment:
Do we have this env var in the real custom app envs? 👀

Reply:
Oh yes, and so so many more. There are ones for streamlit, uvicorn, a generic base URL. We inject quite a few into the variable stack